### DISCUSSION TITLE
Traceback error on select_ortholog.py command with Version 1.1.3

### DISCUSSION THREAD

**Discussion by achoubc** [TheBrownLab/PhyloFisher#49] [Q&A] [answered]
Hello, 

Our server recently updated the program to V1.1.3 and coincidentally (or consequently?) I am getting the following error msg when I do the first round of select_taxa.py and select_ortholog.py. Usually, I do the second round of the two commands to select taxa and an outgroup.

**"Traceback (most recent call last):
 File "/opt/Anaconda3/envs/fisher/bin/select_orthologs.py", line 145, in <module>
  matrix = update_df_taxa(matrix)
 File "/opt/Anaconda3/envs/fisher/bin/select_orthologs.py", line 99, in update_df_taxa
  taxon, _, _, _, include = line.split()
ValueError: too many values to unpack (expected 5)"
I had never have this problem till a couple days ago and can't seem to find it. My command line is "select_orthologs.py --out_group to_include_stram_outgroup.txt"**

The error msg doesn't show if:
1. select_ortholog.py command is executed first then select_taxa.py. If I switch the order of command, the error msg pops up.
2. If I manage to get to the second round of the two commands with the above order, it still throws an error when I get to the select_orthology.py with --out_group option (with or without -c)
3. Update: I think it's a bug with the newer version - If I repeat everything in the V1.1.2, I don't get the error msg.. 

I've been working on the same data/folder with the same config.ini as I was playing around with different taxa subset and -c (gene completeness). It had worked fine before the update. 

Let me know if you need further clarification.
Thanks for your time and developing this magnificent program! 

Anna

**Comment by atice** [comment, marked as answer]
Hi Anna,

Thank you for bringing this to our attention. We changed the output of select_taxa.py to include the long name of all taxa to ease taxa selection and we forgot to consider this in the select_ortholog.py code. It's an easy two-character fix. We will get it fixed and bioconda should have the package rebuilt by Monday. This will be v. 1.2.1. Once you have updated, just rerun both scripts. Thank you for your interest in PhyloFisher.

Alex

Alexander K. Tice, Ph.D.
Department of Biological Sciences
Mississippi State University
027 Harned Hall
Lab Telephone: 662-325-4008

________________________________
From: Anna Cho ***@***.***>
Sent: Friday, October 7, 2022 11:52 AM
To: TheBrownLab/PhyloFisher ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [TheBrownLab/PhyloFisher] Traceback error on select_ortholog.py command with Version 1.1.3 (Discussion #49)


Hello,

Our server recently updated the program to V1.1.3 and coincidentally (or consequently?) I am getting the following error msg when I do the first round of select_taxa.py and select_ortholog.py. Usually, I do the second round of the two commands to select taxa and an outgroup.

"Traceback (most recent call last):
File "/opt/Anaconda3/envs/fisher/bin/select_orthologs.py", line 145, in
matrix = update_df_taxa(matrix)
File "/opt/Anaconda3/envs/fisher/bin/select_orthologs.py", line 99, in update_df_taxa
taxon, _, _, _, include = line.split()
ValueError: too many values to unpack (expected 5)"
I had never have this problem till a couple days ago and can't seem to find it. My command line is "select_orthologs.py --out_group to_include_stram_outgroup.txt"

The error msg doesn't show if:

  1.  select_ortholog.py command is executed first then select_taxa.py. If I switch the order of command, the error msg pops up.
  2.  If I manage to get to the second round of the two commands with the above order, it still throws an error when I get to the select_orthology.py with --out_group option (with or without -c)

I've been working on the same data/folder with the same config.ini as I was playing around with different taxa subset and -c (gene completeness). It had worked fine before the update.

Let me know if you need further clarification.
Thanks for your time and developing program!

Anna

**Reply by achoubc** [reply]
Hello Alex, 

Thanks for the quick response! Really appreciate it. 
I forgot to note that a colleague of mine was having the same error msg but with taxon_collapser.py but I'm guessing it should be resolvable with the new update.. 

Best, 

Anna
